internal/validator/handler.go

```diff
@@ -42,9 +44,49 @@ type handler struct {
 	producer *kafka.Producer
 	schema   *jsonschema.Schema
 	errors   chan error
+	msgChan  chan *kafka.Message
+	vChan    chan messageInfo
 }
 
 func (this *handler) onMessage(ctx context.Context, msg *kafka.Message) {
+	wg.Add(1)
+
+	go func() {
+		defer wg.Done()
+
+		for {
+			if len(this.msgChan) >= cap(this.msgChan) {
+				continue
+			}
+			this.msgChan <- msg
+			break
+		}
+	}()
+	wg.Wait()
+}
+
+func (this *handler) initiateWorkers(
+	ctx context.Context,
+	workers int,
+) {
+	for i := 0; i < workers; i++ {
+		go func() {
+			for {
+				select {
+				case <-ctx.Done():
```

Comment:
This would do the job but it is not very graceful - the workers may be shutdown while there are still requests in flight in channels or workers. Those messages would therefore be lost. A better approach would be:

1. Stop kafka consumer
2. close the msgChan
3. once a worker sees that msgChan is closed it should break out of the loop and exit
4. once all workers are done, close vChan
5. validationProcess in the meantime keeps running, processing the still enqueued messages until it sees that vChan is closed - at that point it exits as all the in-flight messages have been processed
6. at this point there is nothing left in-flight - we can close the producer be done